### DISCUSSION TITLE
Case sensitive search?

### DISCUSSION THREAD

**Discussion by johhansantana** [prisma/prisma#3147] [Q&A] [answered]
Hello, I've seen some issues already posted about this and was wondering if we could have some ideas on a workaround for searching, for example, `firstName` column.

A workaround that I've been thinking of doing is to store all data in lowercase then search using the `contain` where argument

```graphql
{
  users(
    where: {
      AND: [
        {
            firstName: {
              contains: "foo"
            }
          }
      ]
    }
  ) {
    id
    firstName
  }
}
```

But I'm not sure if this is the right solution.

In the sql language we could do something like this

```sql
SELECT * FROM "User"
WHERE lower("firstName") LIKE '%foo%';
```
Is there a way to achieve the latter without using `queryRaw`?

Maybe something like this:
```graphql
{
  users(
    where: {
      AND: [
        {
            firstName: {
              lower: {
                contains: "foo"
              }
            }
          }
      ]
    }
  ) {
    id
    firstName
  }
}
```

**Comment by ryands17** [comment, marked as answer]
Hey @johhansantana 👋 
Unfortunately case-insensitive search is yet to be implemented and we have a [request](https://github.com/prisma/prisma-client-js/issues/690) for the same. Do add a 👍 to the request so that we can judge the priority.

As a workaround, you would need to follow your current implementation of storing the values in lowercase for the time being or use `prisma.queryRaw` for directly using a raw query as mentioned above.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
